src/App.jsx

```diff
@@ -14,12 +14,34 @@ function App() {
       })
   }, [])
 
+  let onlyTags = []
+  if (data) {
+    data.forEach(todo => {
+      todo.tags.forEach(tag => {
+        if (!onlyTags.includes(tag)) {
+          onlyTags.push(tag)
+        }
+      })
+    })
+  }
```

Comment:
This code is working but is also inefficient as you use Array for storing and searching the unique tags. The `includes` method runs for every tag and it searches through the array until it finds the matching element. Imagine there are only 100 todos and each todo has unique tag, that means it will do 1, 2, 3, ... 100 => 5050 steps for 100 rows each with 1 unique tag. When you replace `onlyTags` for `Set` data structure(DS), where `add` operation takes only 1 step, you will improve the code efficiency from 5050 steps to only 100. As you can see, it is important to think in terms of DS as well. I am not going to bore you with more complicated DS but for building this kind of apps you should get familiar with Array, Set and Map.

Reply:
Then what you can do is use your PR and merge with main.  Anyway it is your code logic.  You have exposed me to set data structures.